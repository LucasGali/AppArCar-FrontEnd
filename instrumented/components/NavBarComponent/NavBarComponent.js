import React from 'react';
import { Link } from 'react-router-dom';
import { useHistory } from 'react-router-dom';
import iconSet from 'icons.json';
import IcomoonReact from 'icomoon-react';
import text from 'utils/text';
import './NavBarComponent.css';
import { useDispatch, connect } from 'react-redux';
import { userData } from '../../actions/index';
import { SIGN_OFF } from '../../actions/types';

const NavBarComponent = (props) => {
  let history = useHistory();
  const dispatch = useDispatch();
  const GuestNavBar = () => {
    return (
      <nav className="navbar" data-testid="navbar">
        <div>
          <IcomoonReact
            className="icons-nav-user"
            iconSet={iconSet}
            color="white"
            size={30}
            icon="location"
          />
          <span>{text.navbar.app}</span>
        </div>
        <div>
          <button
            className="btn-sigin-signup btn  my-2 my-sm-0"
            onClick={() => history.push('/signin')}
            data-testid="signin-button"
          >
            {text.navbar.signin}
          </button>
          <button
            className="btn btn-sigin-signup my-2 my-sm-0 "
            onClick={() => history.push('/signup')}
          >
            {text.navbar.signup}
          </button>
        </div>
      </nav>
    );
  };

  const Registration = () => {
    return (
      <nav className="navbar">
        <div>
          <IcomoonReact
            className="icons-nav-user"
            iconSet={iconSet}
            color="white"
            size={30}
            icon="location"
          />
          <span>{text.navbar.app}</span>
        </div>
        <div>
          <button
            className="btn btn-sigin-signup my-2 my-sm-0 "
            onClick={() => history.push('/')}
          >
            <span>{text.navbar.back}</span>
          </button>
        </div>
      </nav>
    );
  };

  const signOff = () => {
    localStorage.clear();
    dispatch(userData(SIGN_OFF));
    history.push('/');
  };

  const UserNavBar = () => {
    return (
      <nav className="navbar">
        <div>
          <IcomoonReact
            className="icons-nav-user"
            iconSet={iconSet}
            color="white"
            size={30}
            icon="location"
          />
          <span>AppArCar</span>
        </div>
        <div className="nav-wrapper-icons" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link to="/Home" className="nav-link text-blue">
                <span title={text.navbar.home}>
                  <IcomoonReact
                    className="icons-nav-user"
                    iconSet={iconSet}
                    color="white"
                    size={30}
                    icon="home"
                  />
                </span>
              </Link>
            </li>
            <li className="nav-item">
              <Link
                id="profileBtn"
                to="/Profile"
                className="nav-link text-blue"
                data-testid="profile-icon"
              >
                <span title={text.navbar.profile}>
                  <IcomoonReact
                    className="icons-nav-user"
                    iconSet={iconSet}
                    color="white"
                    size={30}
                    icon="user"
                  />
                </span>
              </Link>
            </li>
            <li className="nav-item">
              <Link
                to="/Metrics"
                className="nav-link text-blue"
                data-testid="metrics-icon"
              >
                <span title={text.navbar.metrics}>
                  <IcomoonReact
                    className="icons-nav-user"
                    iconSet={iconSet}
                    color="white"
                    size={30}
                    icon="stats-dots"
                  />
                </span>
              </Link>
            </li>
            <li className="nav-item sign-off">
              <button
                className="btn my-2 my-sm-0 "
                data-testid="signOffBtn"
                onClick={signOff}
              >
                <span title={text.navbar.signOff}>
                  <IcomoonReact
                    className="icons-nav-user"
                    iconSet={iconSet}
                    color="white"
                    size={30}
                    icon="exit"
                  />
                  {text.navbar.signOff}
                </span>
              </button>
            </li>
          </ul>
        </div>
      </nav>
    );
  };

  const NavBar = () => {
    return props.isGuest ? (
      <GuestNavBar />
    ) : props.inRegistration ? (
      <Registration />
    ) : (
      <UserNavBar />
    );
  };

  return (
    <div>
      <NavBar />
    </div>
  );
};

const mapStateToProps = (state) => ({
  storedUser: state.userData,
});

export default connect(mapStateToProps)(NavBarComponent);
