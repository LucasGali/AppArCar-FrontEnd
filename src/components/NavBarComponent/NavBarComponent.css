.navbar {
  background-color: black;
  color: white;
}

.navbar span {
  margin-left: 10px;
}

.nav-wrapper-icons ul {
  display: flex;
  flex-direction: row;
}

.nav-wrapper-icons ul li {
  margin-left: 10px;
}

.navbar button {
  color: white;
}

.navbar button:hover {
  color: bisque;
}

button.btn-sigin-signup {
  border: 1px solid black;
  margin-right: 5px;
}

.sign-off {
  display: flex;
  align-self: flex-end;
}

@media only screen and (max-height: 900px) {
  .navbar {
    padding: 0rem 1rem;
  }
  .nav-item span .icons-nav-user {
    height: 25px;
    width: 25px;
  }
}
