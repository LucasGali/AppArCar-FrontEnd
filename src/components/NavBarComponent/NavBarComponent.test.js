import { render, screen } from '@testing-library/react';
import NavBarComponent from './NavBarComponent';
import { MemoryRouter } from 'react-router-dom';

describe('NavBarComponent', () => {
  it('renders default form acording to snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBarComponent />
      </MemoryRouter>,
    );
    expect(asFragment(<NavBarComponent />)).toMatchSnapshot();
  });
});
