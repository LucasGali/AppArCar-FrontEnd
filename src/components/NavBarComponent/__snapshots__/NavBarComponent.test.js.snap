// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarComponent renders default form acording to snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar"
      data-testid="navbar"
    >
      <div>
        <svg
          class="icons-nav-user"
          height="30"
          style="display: inline-block; vertical-align: middle;"
          viewBox="0 0 1024 1024"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 0c-176.732 0-320 143.268-320 320 0 320 320 704 320 704s320-384 320-704c0-176.732-143.27-320-320-320zM512 512c-106.040 0-192-85.96-192-192s85.96-192 192-192 192 85.96 192 192-85.96 192-192 192z"
            style="fill: white;"
          />
        </svg>
        <span>
          AppArCar
        </span>
      </div>
      <div>
        <button
          class="btn-sigin-signup btn  my-2 my-sm-0"
          data-testid="signin-button"
        >
          Ingresar
        </button>
        <button
          class="btn btn-sigin-signup my-2 my-sm-0 "
        >
          Registrarse
        </button>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
